2024-06-28 10:49:53,539 INFO    StreamThr :11302 [internal.py:wandb_internal():85] W&B internal server running at pid: 11302, started at: 2024-06-28 10:49:53.538438
2024-06-28 10:49:53,541 DEBUG   HandlerThread:11302 [handler.py:handle_request():158] handle_request: status
2024-06-28 10:49:53,545 INFO    WriterThread:11302 [datastore.py:open_for_write():87] open: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/run-mh0ov8ap.wandb
2024-06-28 10:49:53,546 DEBUG   SenderThread:11302 [sender.py:send():379] send: header
2024-06-28 10:49:53,561 DEBUG   SenderThread:11302 [sender.py:send():379] send: run
2024-06-28 10:49:54,215 INFO    SenderThread:11302 [dir_watcher.py:__init__():211] watching files in: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files
2024-06-28 10:49:54,215 INFO    SenderThread:11302 [sender.py:_start_run_threads():1188] run started: mh0ov8ap with start time 1719551993.538537
2024-06-28 10:49:54,222 DEBUG   HandlerThread:11302 [handler.py:handle_request():158] handle_request: check_version
2024-06-28 10:49:54,223 DEBUG   SenderThread:11302 [sender.py:send_request():406] send_request: check_version
2024-06-28 10:49:54,367 DEBUG   HandlerThread:11302 [handler.py:handle_request():158] handle_request: run_start
2024-06-28 10:49:54,369 ERROR   HandlerThread:11302 [gpu.py:is_available():381] Error initializing NVML: Driver Not Loaded
2024-06-28 10:49:54,369 DEBUG   HandlerThread:11302 [system_info.py:__init__():26] System info init
2024-06-28 10:49:54,369 DEBUG   HandlerThread:11302 [system_info.py:__init__():41] System info init done
2024-06-28 10:49:54,369 INFO    HandlerThread:11302 [system_monitor.py:start():194] Starting system monitor
2024-06-28 10:49:54,370 INFO    SystemMonitor:11302 [system_monitor.py:_start():158] Starting system asset monitoring threads
2024-06-28 10:49:54,370 INFO    HandlerThread:11302 [system_monitor.py:probe():214] Collecting system info
2024-06-28 10:49:54,370 INFO    SystemMonitor:11302 [interfaces.py:start():188] Started cpu monitoring
2024-06-28 10:49:54,371 INFO    SystemMonitor:11302 [interfaces.py:start():188] Started disk monitoring
2024-06-28 10:49:54,371 INFO    SystemMonitor:11302 [interfaces.py:start():188] Started memory monitoring
2024-06-28 10:49:54,372 INFO    SystemMonitor:11302 [interfaces.py:start():188] Started network monitoring
2024-06-28 10:49:54,396 DEBUG   HandlerThread:11302 [system_info.py:probe():152] Probing system
2024-06-28 10:49:54,400 DEBUG   HandlerThread:11302 [system_info.py:_probe_git():137] Probing git
2024-06-28 10:49:54,408 DEBUG   HandlerThread:11302 [system_info.py:_probe_git():145] Probing git done
2024-06-28 10:49:54,409 DEBUG   HandlerThread:11302 [system_info.py:probe():200] Probing system done
2024-06-28 10:49:54,409 DEBUG   HandlerThread:11302 [system_monitor.py:probe():223] {'os': 'Linux-5.4.0-187-generic-x86_64-with-glibc2.31', 'python': '3.9.19', 'heartbeatAt': '2024-06-28T05:19:54.396565', 'startedAt': '2024-06-28T05:19:53.526509', 'docker': None, 'cuda': None, 'args': ('-data', 'BloodMNIST', '-m', 'resnet18', '--algorithm', 'FedAvg', '-gr', '100', '-did', '0'), 'state': 'running', 'program': '/home/student/R/fl/PFLlib/system/main.py', 'codePathLocal': 'main.py', 'codePath': 'system/main.py', 'git': {'remote': 'https://github.com/TsingZ0/PFLlib.git', 'commit': '47f052069dcd19fbb7e37b14dd37823a363c2464'}, 'email': 'sachugowda@hotmail.com', 'root': '/home/student/R/fl/PFLlib', 'host': 'cse-HP-EliteDesk-800-G1-TWR', 'username': 'student', 'executable': '/usr/bin/python3', 'cpu_count': 4, 'cpu_count_logical': 8, 'cpu_freq': {'current': 2481.4116249999997, 'min': 800.0, 'max': 4000.0}, 'cpu_freq_per_core': [{'current': 2485.622, 'min': 800.0, 'max': 4000.0}, {'current': 2541.838, 'min': 800.0, 'max': 4000.0}, {'current': 2572.049, 'min': 800.0, 'max': 4000.0}, {'current': 2396.682, 'min': 800.0, 'max': 4000.0}, {'current': 2424.389, 'min': 800.0, 'max': 4000.0}, {'current': 2510.956, 'min': 800.0, 'max': 4000.0}, {'current': 2482.956, 'min': 800.0, 'max': 4000.0}, {'current': 2436.801, 'min': 800.0, 'max': 4000.0}], 'disk': {'/': {'total': 258.4741439819336, 'used': 184.05631256103516}}, 'memory': {'total': 15.53030014038086}}
2024-06-28 10:49:54,409 INFO    HandlerThread:11302 [system_monitor.py:probe():224] Finished collecting system info
2024-06-28 10:49:54,409 INFO    HandlerThread:11302 [system_monitor.py:probe():227] Publishing system info
2024-06-28 10:49:54,409 INFO    HandlerThread:11302 [system_monitor.py:probe():229] Finished publishing system info
2024-06-28 10:49:55,221 INFO    Thread-12 :11302 [dir_watcher.py:_on_file_created():271] file/dir created: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files/wandb-metadata.json
2024-06-28 10:49:55,223 INFO    WriterThread:11302 [datastore.py:close():296] close: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/run-mh0ov8ap.wandb
2024-06-28 10:49:55,363 INFO    SenderThread:11302 [sender.py:finish():1608] shutting down sender
2024-06-28 10:49:55,364 INFO    SenderThread:11302 [dir_watcher.py:finish():358] shutting down directory watcher
2024-06-28 10:49:56,222 INFO    SenderThread:11302 [dir_watcher.py:finish():388] scan: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files
2024-06-28 10:49:56,223 INFO    SenderThread:11302 [dir_watcher.py:finish():402] scan save: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files/wandb-metadata.json wandb-metadata.json
2024-06-28 10:49:56,223 INFO    SenderThread:11302 [dir_watcher.py:finish():402] scan save: /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files/config.yaml config.yaml
2024-06-28 10:49:56,231 INFO    SenderThread:11302 [file_pusher.py:finish():169] shutting down file pusher
2024-06-28 10:49:56,232 INFO    SenderThread:11302 [file_pusher.py:join():175] waiting for file pusher
2024-06-28 10:49:57,251 INFO    wandb-upload_0:11302 [upload_job.py:push():130] Uploaded file /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files/wandb-metadata.json
2024-06-28 10:49:57,297 INFO    wandb-upload_1:11302 [upload_job.py:push():130] Uploaded file /home/student/R/fl/PFLlib/system/wandb/run-20240628_104953-mh0ov8ap/files/config.yaml
2024-06-28 10:49:57,734 INFO    SenderThread:11302 [file_stream.py:finish():601] file stream finish called
2024-06-28 10:49:58,085 INFO    SenderThread:11302 [file_stream.py:finish():605] file stream finish is done
2024-06-28 10:49:58,087 ERROR   StreamThr :11302 [internal.py:wandb_internal():173] Thread HandlerThread:
Traceback (most recent call last):
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/internal_util.py", line 48, in run
    self._run()
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/internal_util.py", line 99, in _run
    self._process(record)
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/internal.py", line 278, in _process
    self._hm.handle(record)
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/handler.py", line 150, in handle
    handler(record)
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/handler.py", line 160, in handle_request
    handler(record)
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/handler.py", line 704, in handle_request_run_start
    self._tb_watcher = tb_watcher.TBWatcher(
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/internal/tb_watcher.py", line 126, in __init__
    wandb.tensorboard.reset_state()
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/lib/lazyloader.py", line 57, in __getattr__
    module = self._load()
  File "/home/student/.local/lib/python3.9/site-packages/wandb/sdk/lib/lazyloader.py", line 32, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/student/.local/lib/python3.9/site-packages/wandb/integration/tensorboard/__init__.py", line 3, in <module>
    from .log import _log, log, reset_state, tf_summary_to_dict  # noqa: F401
  File "/home/student/.local/lib/python3.9/site-packages/wandb/integration/tensorboard/log.py", line 35, in <module>
    Summary = pb.Summary if pb else None
  File "/home/student/.local/lib/python3.9/site-packages/wandb/util.py", line 219, in __getattribute__
    state.load()
  File "/home/student/.local/lib/python3.9/site-packages/wandb/util.py", line 212, in load
    self.module.__spec__.loader.exec_module(self.module)
  File "/home/student/.local/lib/python3.9/site-packages/tensorboard/compat/proto/summary_pb2.py", line 17, in <module>
    from tensorboard.compat.proto import tensor_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__pb2
  File "/home/student/.local/lib/python3.9/site-packages/tensorboard/compat/proto/tensor_pb2.py", line 16, in <module>
    from tensorboard.compat.proto import resource_handle_pb2 as tensorboard_dot_compat_dot_proto_dot_resource__handle__pb2
  File "/home/student/.local/lib/python3.9/site-packages/tensorboard/compat/proto/resource_handle_pb2.py", line 16, in <module>
    from tensorboard.compat.proto import tensor_shape_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__shape__pb2
  File "/home/student/.local/lib/python3.9/site-packages/tensorboard/compat/proto/tensor_shape_pb2.py", line 36, in <module>
    _descriptor.FieldDescriptor(
  File "/home/student/.local/lib/python3.9/site-packages/google/protobuf/descriptor.py", line 553, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
TypeError: Descriptors cannot be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
